File: src/main/java/me/aikoo/Karl/Interpreter/parser/ast/values/IntValue.java
package me.aikoo.Karl.Interpreter.parser.ast.values;

import me.aikoo.Karl.Interpreter.parser.TokenType;

public class IntValue extends Value {
    private final int value;

    public IntValue(int value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return Integer.toString(value);
    }

    @Override
    public int toInt() {
        return value;
    }

    @Override
    public float toFloat() {
        return Float.parseFloat(toString());
    }

    @Override
    public TokenType getType() {
        return TokenType.INT_VALUE;
    }
}

File: src/main/kotlin/studio/karllang/karl/parser/ast/values/IntValue.kt
package studio.karllang.karl.parser.ast.values

import studio.karllang.karl.lexer.TokenType

class IntValue(private val value: Int) : Value() {
    override val type: TokenType = TokenType.INT_VALUE

    override fun toString(): String {
        return value.toString()
    }

    override fun toInt(): Int {
        return value
    }

    override fun toFloat(): Float {
        return toString().toFloat()
    }
}